### ISSUE TITLE
Duplicate code

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/greenhouse-effect#337]
For https://github.com/phetsims/greenhouse-effect/issues/331 ...

> - [ ] Are any significant chunks of code duplicated? In addition to manual identification, tools include: WebStorm _Code > Analyze Code > Locate Duplicates_ and https://github.com/danielstjules/jsinspect.

I used WebStorm _Code > Analyze Code > Locate Duplicates_, and ignored duplicates in js/micro/.  Here are the duplicates that seemed significant.

---

- [x] `energyPacketCrossedAltitude ` in FluxSensor.ts and EnergyAbsorbingEmittingLayer.ts

```ts
private energyPacketCrossedAltitude( energyPacket: EMEnergyPacket ): boolean {
  const altitude = this.altitudeProperty.value;
  return ( energyPacket.previousAltitude > altitude && energyPacket.altitude <= altitude ) ||
         ( energyPacket.previousAltitude < altitude && energyPacket.altitude >= altitude );
}
```

- [x] `new ArrowNode` in FluxMeterNode.ts:

```ts
this.downArrow = new ArrowNode(
  boundsRectangle.width / 2,
  boundsRectangle.height / 2,
  boundsRectangle.width / 2,
  boundsRectangle.height / 2,
  options.arrowNodeOptions
);
this.upArrow = new ArrowNode(
  boundsRectangle.width / 2,
  boundsRectangle.height / 2,
  boundsRectangle.width / 2,
  boundsRectangle.height / 2,
  options.arrowNodeOptions
);
```

- [x] `const options: PanelOptions` in InfraRedPanel.ts and SunlightPanel.ts

```ts
const options: PanelOptions = {

  minWidth: width,
  maxWidth: width,
  xMargin: PANEL_MARGIN,
  yMargin: PANEL_MARGIN,
  align: 'center' as const,
  fill: GreenhouseEffectColors.controlPanelBackgroundColorProperty,

  // pdom
  tagName: 'div',
  labelTagName: 'h3',
  labelContent: GreenhouseEffectStrings.infraredStringProperty,

  // phet-io
  tandem: tandem,
  visiblePropertyOptions: {
    phetioFeatured: true
  }
};
```

- [x] `isInfrared ` in Photon.ts, EMEnergyPacket.ts, Wave.ts

```ts
/**
 * convenience method for determining whether this is an infrared photon
 */
public get isInfrared(): boolean {
  return this.wavelength === GreenhouseEffectConstants.INFRARED_WAVELENGTH;
}
```

- [x] `isVisible` in Photon.ts, EMEnergyPacket.ts, Wave.ts

```ts
/**
 * convenience method for determining whether the EM energy contained in this packet is in the visible light range
 */
public get isVisible(): boolean {
  return this.wavelength === GreenhouseEffectConstants.VISIBLE_WAVELENGTH;
}
```

**Comment by jbphet** [comment]
I've addressed all items listed above.  They all felt pretty good except for the last two, which relate to testing instances of types `Photon`, `Wave`, and `EMEnergyPacket` for whether they are visible or infrared.  I thought about trying to use a mixin, but have found these to be tricky and hard to maintain, so I ended up going with a pair of simple utility functions.  This particular change didn't feel like a _big_ win, but at least it consolidates it all into one place which would make it easier to change if necessary.

@pixelzoom - feel free to take a look if you see this going by, but I think I've addressed them all, so I'll go ahead and close.
